README.md

```diff
@@ -204,6 +204,31 @@ func init() {
 }
 ```
 
+**Logger**
+
+xray uses an interface for its logger:
+
+```go
+type Logger interface {
+  Log(level LogLevel, msg fmt.Stringer)
+}
+
+const (
+  LogLevelDebug LogLevel = iota + 1
+  LogLevelInfo
+  LogLevelWarn
+  LogLevelError
+)
+```
+
+The default logger logs to stdout at "info" and above. To change the logger, call `xray.SetLogger(myLogger)`. There is a default logger implementation that writes to an `io.Writer` from a specified minimum log level. For example, to log to stderr at "error" and above:
+
+```go
+xray.SetLogger(xraylog.NewDefaultLogger(os.Stderr, xraylog.LogLevelError))
```

Comment:
I would suggest you could include the logger setting into `xray.Config{}` to keep consistency. It can be inconvenient for customers to configure setting through different way.

Reply:
I wanted to keep it in the config, but I decided it was not safe. The logger is something you only want to set up once, but the config can be changed many times. In fact, you can set the config in the context using `ContextWithConfig`, so you can have multiple segments at the same time using different configs. Since the logger is global, you can't have two different logging configs at the same time. Therefore, to make sure it is safe and avoid race conditions, the logger must be set up one time only in `init()` method.

Reply:
Hi @muirrn ,
I would still prefer having logger setting into  `xray.Config{}`. This adds flexibility for users if they want logger per context or at global level. This also eliminates new method to be introduced for configuring logger apart from current way of  `xray.Config{}` and helps in achieving both, per context as well as global level logging functionality. For example [`BeginSegment`](https://github.com/aws/aws-xray-sdk-go/blob/9779f16ea01fdfc11c54971ed8547dbaa442bb42/xray/segment.go#L59) accepts context struct, and if present, configuration is applied from that context else from global config. Could you explain more on safety and race conditions issues with this approach, since logger instance would be created/accessed per context?

Reply:
Hi @muirrn ,
Thank you for the explanation. I just wanted to make sure that we think through all scenarios. Your observation is correct about seelog usage.